src/ui/chat-window.js

```diff
@@ -497,14 +498,23 @@ export class ChatScene extends Phaser.Scene {
                 (event.keyCode <= 222 && event.keyCode >= 219)
             ) {
                 // arbitrary input length limit
-                if (this.userMessage.text.length < 55) {
+                if (
+                    this.userMessage.text.length < 55 &&
+                    this.recentKeys.get(event.keyCode) == undefined
+                ) {
                     this.userMessage.text =
                         this.userMessage.text.substr(
                             0,
                             this.userMessage.text.length - 1
                         ) +
                         event.key +
                         "*";
+
+                    // Phaser has an input bug where pressing multiple keys simultaneously
+                    // (or near-simultaneously) results in duplicate letters being detected as input.
+                    // Here we add recently typd letters to a decaying list so that they aren't duplicated
+                    // within 100 milliseconds
+                    this.recentKeys.put(event.keyCode, 100);
```

Comment:
```suggestion
                    this.recentKeys.put(event.keyCode, 10);
```
Played around with it and I think we can lower it a lot and basically have no drawbacks as long as it's non-zero

Reply:
Ah nice nice. Updated!